[BUG] 428 should be an allowed negative status

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Negative tests fail with 
```
  E   1. Accepted negative data
  E   
  E       Allowed statuses: 400, 401, 403, 404, 422, 5xx
  E   
  E   [428] Precondition Required:
  E   
  E       `{"error":{"code":1428,"message":"If-Match header not provided."}}`
  E   
  E   Reproduce with: 
  E   
  E       curl -X PATCH -H 'host: mockserver:1234' -H 'authorization: ***"is_admin": true, "id": "admin", "name": "Admin Doe", "first_name": "Admin", "last_name": "Doe", "email": "admin.doe@gmail.com", "full_name": "Admin Doe"}' -H 'content-type: application/json' -d '[]' http://localhost/api/data/platform/config
  E   
  E   ====================
```
but 428 is a valid status in this case, as the apispec in this case [expects an If-Match header](https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/platform/api.spec.yaml#L31) 


### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

Apispec:
```
openapi: 3.0.2
paths:
  /platform/config:
    patch:
      parameters:
        - $ref: "#/components/parameters/If-Match"
      requestBody:
        required: true
        content:
         application/json:
              schema:
                $ref: "#/components/schemas/PlatformConfigPatch"
      responses:
        "200":
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/PlatformConfig"
        default:
          $ref: "#/components/responses/Error"
      tags:
        - platform
components:
  schemas:
    PlatformConfig:
      type: object
      properties:
        etag:
          $ref: "#/components/schemas/ETag"
        incident_banner:
          $ref: "#/components/schemas/IncidentBanner"
      required:
        - etag
        - incident_banner
      additionalProperties: false
    PlatformConfigPatch:
      type: object
      properties:
        incident_banner:
          $ref: "#/components/schemas/IncidentBanner"
      additionalProperties: false
    ETag:
      type: string
      example: "9EE498F9D565D0C41E511377425F32F3"
    IncidentBanner:
      type: string
    ErrorResponse:
      type: object
      properties:
        error:
          type: object
          properties:
            message:
              type: string
          required:
            - "message"
      required:
        - "error"
  responses:
    Error:
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/ErrorResponse"
  parameters:
    If-Match:
      in: header
      name: If-Match
      required: true
      schema:
        $ref: "#/components/schemas/ETag"
```

### Expected behavior
To accept 428 to be a valid failure status.

Alternatively, to be able to configure this in `schemathesis.from_pytest_fixture`

### Environment

```
- OS: Linux
- Python version: 3.12
- Schemathesis version: 3.39.5
- Spec version: Open API 3.0.2
```

### Additional context

[Relevant PR](https://github.com/SwissDataScienceCenter/renku-data-services/pull/599)
[Relevant test](https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/test/bases/renku_data_services/data_api/test_schemathesis.py)

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -4,6 +4,10 @@ Changelog
 :version:`Unreleased <v3.39.5...HEAD>` - TBD
 --------------------------------------------
 
+**Changed**
+
+- Add HTTP 428 status to the allowed status list of the ``negative_data_rejection`` check. :issue:`2669`
+
 .. _v3.39.5:
 
 :version:`3.39.5 <v3.39.4...v3.39.5>` - 2024-12-27
@@ -1700,7 +1704,7 @@ After:
 
 - CLI: Warning if the API returns too many HTTP 401.
 - Add ``SCHEMATHESIS_BASE_URL`` environment variable for specifying ``--base-url`` in CLI.
-- Collect anonymyzed CLI usage telemetry when reports are uploaded. We do not collect any free-form values you use in your CLI,
+- Collect anonymized CLI usage telemetry when reports are uploaded. We do not collect any free-form values you use in your CLI,
   except for header names. Instead, we measure how many times you use each free-form option in this command.
   Additionally we count all non-default hook types only by hook name.
 

--- a/src/schemathesis/internal/checks.py
+++ b/src/schemathesis/internal/checks.py
@@ -21,7 +21,7 @@ CheckFunction = Callable[["CheckContext", "GenericResponse", "Case"], Optional[b
 @dataclass
 class NegativeDataRejectionConfig:
     # 5xx will pass through
-    allowed_statuses: list[str] = field(default_factory=lambda: ["400", "401", "403", "404", "422", "5xx"])
+    allowed_statuses: list[str] = field(default_factory=lambda: ["400", "401", "403", "404", "422", "428", "5xx"])
 
 
 @dataclass

